### PR TITLE
Check for update

### PR DESCRIPTION
**Pull request by alex-oreshko**
Some more extension

### CODE REVIEW

File: AppCenterAssets/AppCenterAssets/Internals/Model/MSAssetsConfiguration.h [hunk 1]
```diff
@@ -8,12 +8,12 @@
 /**
  * The version of the app that was deployed (for a native app upgrade).
  */
-@property(nonatomic, copy, nonnull) NSString *appVersion;
+@property(nonatomic, copy, nullable) NSString *appVersion;
```
Comment (annakocheshkova) [inline AppCenterAssets/AppCenterAssets/Internals/Model/MSAssetsConfiguration.h]: Why `nullable`? These properties can not be null.

File: AppCenterAssets/AppCenterAssets/MSAssetsDelegate.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+@protocol MSAssetsDelegate <NSObject>
+
+@optional
+
+/**
+ * Callback method that will be called by CheckForUpdate
+ *
+ * @param package The instance of MSRemotePackage.
+ */
+- (void)packageForUpdate:(MSRemotePackage *)package;
```
Comment (annakocheshkova) [inline AppCenterAssets/AppCenterAssets/MSAssetsDelegate.h]: I think you should also implement `didFailToQueryPackage:(NSError *)error` (f. e. how it's done in `MSAnalyticsDelegate`).

File: AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m [hunk 3]
```diff
@@ -1,24 +1,27 @@
 #import "MSAssets.h"
 #import "MSAssetsUpdateState.h"
 #import "MSLocalPackage.h"
+#import <UIKit/UIKit.h>
```
Comment (max-mironov) [inline AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m]: Do we need this import?

File: AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m [hunk 4]
```diff
@@ -28,13 +31,54 @@ - (MSRemotePackage *)checkForUpdate:(NSString *)deploymentKey {
...
+
+        if (!update || update.updateAppVersion ||
+            (localPackage && ([update.packageHash isEqualToString:localPackage.packageHash])) ||
+            ((!localPackage || localPackage.isDebugOnly) && [config.packageHash isEqualToString:update.packageHash] )){
+
+            if (update && update.updateAppVersion){
+                NSLog(@"An update is available but it is not targeting the binary version of your app.");
+
+                if ([[self delegate] respondsToSelector:@selector(packageForUpdate:)])
+                    [[self delegate] packageForUpdate:nil];
```
Comment (annakocheshkova) [inline AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m]: Here `didFailToQueryPackage` would be suitable. User that has subscribed to `packageForUpdate` callback expects it to be called only if a real package has come.

File: AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m [hunk 5]
```diff
@@ -28,13 +28,54 @@ - (MSRemotePackage *)checkForUpdate:(NSString *)deploymentKey {
...
+            NSLog(@"#Error: %@", error.localizedDescription);
+        };
+        MSRemotePackage *update = [package mutableCopy];
+
+        if (!update || update.updateAppVersion ||
+            (localPackage && ([update.packageHash isEqualToString:localPackage.packageHash])) ||
+            ((!localPackage || localPackage.isDebugOnly) && [config.packageHash isEqualToString:update.packageHash] )){
+
+            if (update && update.updateAppVersion){
+                NSLog(@"An update is available but it is not targeting the binary version of your app.");
```
[the remark this answers is not in the document]
Comment (alex-oreshko) [reply AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m]: I've added callback with single param (package), it seems needs one more param (error)

File: AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m (hunk 3 above)
Comment (alex-oreshko) [reply AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m]: Yes I need it to execute this
    configuration.clientUniqueId = [[[UIDevice currentDevice] identifierForVendor] UUIDString];

Comment (max-mironov) [reply AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m]: Gotcha!

Comment (alex-oreshko) [reply AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m]: May be will refactor later

File: AppCenterAssets/AppCenterAssets/MSAssetsDelegate.h (hunk 2 above)
Comment (alex-oreshko) [reply AppCenterAssets/AppCenterAssets/MSAssetsDelegate.h]: you mean, don't add error param to didReceiveRemotePackageOnUpdateCheck, but add didFailToQueryPackage:(NSError *)error ?

File: AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m (hunk 4 above)
Comment (alex-oreshko) [reply AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m]: okay

**Pull request merged**